tests/browser_context_test.go

```diff
@@ -750,119 +745,50 @@ func TestBrowserContextWaitForEvent(t *testing.T) {
 			ctx, cancel := context.WithTimeout(tb.context(), 5*time.Second)
 			defer cancel()
 
-			var p1, p2 *common.Page
-			// We need to run waitForEvent in parallel to the page creation.
-			// If we run them synchronously then waitForEvent will wait
-			// indefinitely and eventually the test will timeout.
-			err = tb.run(
-				ctx,
-				func() error {
-					op := optsOrPredicateToGojaValue(t, tb, tc.optsOrPredicate)
-					resp, err := bc.WaitForEvent(tc.event, op)
-					if resp != nil {
-						var ok bool
-						p1, ok = resp.(*common.Page)
-						require.True(t, ok)
-					}
-					return err
-				},
-				func() error {
-					var err error
-					p2, err = bc.NewPage()
-					return err
-				},
+			var (
+				aboutToCallWait = make(chan bool)
+				waitDone        = make(chan bool)
+				p1ID, p2ID      string
 			)
 
-			if tc.wantErr == "" {
-				assert.NoError(t, err)
-				// We want to make sure that the page that was created with
-				// newPage matches the return value from waitForEvent.
-				assert.Equal(t, p1.MainFrame().ID(), p2.MainFrame().ID())
-				return
-			}
-
-			assert.ErrorContains(t, err, tc.wantErr)
-		})
-	}
-}
+			go func() {
+				defer close(waitDone)
 
-// optsOrPredicate is a helper type to enable us to package up the optional
-// arguments which could either be a predicate function, or the options object
-// which contains a predicate function and a timeout.
-type optsOrPredicate struct {
-	predicate     *string
-	timeout       *int64
-	justPredicate *string
-}
-
-func stringPtr(value string) *string {
-	strPointer := new(string)
-	*strPointer = value
-	return strPointer
-}
-
-func int64Ptr(value int64) *int64 {
-	int64Pointer := new(int64)
-	*int64Pointer = value
-	return int64Pointer
-}
+				var resp any
+				close(aboutToCallWait)
```

Comment:
I understand we use this channel to guarantee that `bc.NewPage()` is called after `bc.WaitForEvent`, is that right? If that is the case, should we close it after actually calling `waitForEvent`?